src/Orleans/Async/TaskUtility.cs

```diff
@@ -0,0 +1,55 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+namespace Orleans.Async
+{
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Extension methods for <see cref="Task"/>.
+    /// </summary>
+    public static class TaskUtility
+    {
+        /// <summary>
+        /// Returns a faulted task.
+        /// </summary>
+        /// <param name="exception">The exception which the return task faulted with.</param>
+        /// <returns>A faulted task.</returns>
+        public static Task<object> Faulted(Exception exception)
+        {
+            var completion = new TaskCompletionSource<object>();
+            completion.TrySetException(exception);
+            return completion.Task;
+        }
+
+        /// <summary>
+        /// Returns a completed task.
+        /// </summary>
+        /// <returns>A completed task.</returns>
+        public static Task<object> Completed()
+        {
+            return Task.FromResult((object)null);
```

Comment:
Not at my PC, but I think Task.Done returns Task, not Task<object>

-----Original Message-----
From: "Dmitry Vakulenko" <notifications@github.com>
Sent: ‎10/‎22/‎2015 8:56 PM
To: "dotnet/orleans" <orleans@noreply.github.com>
Cc: "Reuben Bond" <reuben.bond@gmail.com>
Subject: Re: [orleans] Feature: Runtime Code Generation (#528)

In src/Orleans/Async/TaskUtility.cs:
> +        /// <param name="exception">The exception which the return task faulted with.</param>
> +        /// <returns>A faulted task.</returns>
> +        public static Task<object> Faulted(Exception exception)
> +        {
> +            var completion = new TaskCompletionSource<object>();
> +            completion.TrySetException(exception);
> +            return completion.Task;
> +        }
> +
> +        /// <summary>
> +        /// Returns a completed task.
> +        /// </summary>
> +        /// <returns>A completed task.</returns>
> +        public static Task<object> Completed()
> +        {
> +            return Task.FromResult((object)null);
Why not use Task.Done from TaskExtensions?
—
Reply to this email directly or view it on GitHub.